File: app/src/main/java/pl/rpieja/flat/dto/Expense.java
package pl.rpieja.flat.dto;

import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.Date;
import java.util.List;

/**
 * Created by radix on 2017-11-01.
 */

public class Expense implements ChargeLike {
    public String name, rawAmount;
    public Date date;
    public int id;
    public List<User> to;
    public User from;
    public double amount;

    @Nullable
    @Override
    public Double getChargeAmount() {
        return amount;
    }

    @Nullable
    @Override
    public String getChargeName() {
        return name;
    }

    @Nullable
    @Override
    public List<User> getFromUsers() {
        return Collections.singletonList(from);
    }

    @Nullable
    @Override
    public List<User> getToUsers() {
        return to;
    }
}

File: app/src/main/java/pl/rpieja/flat/dto/Expense.kt
package pl.rpieja.flat.dto

import pl.memleak.flat.fragment.ExpenseFragment
import java.util.*

data class Expense(
        var id: String,
        var name: String,
        var date: Date,
        var to: List<User>,
        var from: User,
        var amount: Double,
        var category: Category?
) : ChargeLike {

    constructor(obj: ExpenseFragment) : this(
            id = obj.id(),
            name = obj.name(),
            date = obj.date(),
            to = obj.toUsers()?.map { User(it.fragments().userFragment()) }.orEmpty(),
            from = User(obj.fromUser().fragments().userFragment()),
            amount = obj.amount()!!,
            category = obj.category()?.fragments()?.categoryFragment()?.let { Category(it) }
    )

    override val chargeAmount: Double
        get() = amount
    override val chargeName: String
        get() = name
    override val chargeCategory: String
        get() = category?.name ?: ""
    override val fromUsers: List<User>
        get() = listOf(from)
    override val toUsers: List<User>
        get() = to
}
